### PR TITLE
First Pass on Capsules (Docker only)

### PR DESCRIPTION
**Pull request by rocketbitz**
98% coverage baby 🔥 
```
ok      github.com/caesium-dev/caesium/internal/capsule/docker  (cached)        coverage: 98.0% of statements
```

### CODE REVIEW

File: internal/capsule/docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	ContainerRemove(context.Context, string, types.ContainerRemoveOptions) error
+	ContainerLogs(context.Context, string, types.ContainerLogsOptions) (io.ReadCloser, error)
+}
+
+var (
+	stateMap = map[string]capsule.State{
+		"created":    capsule.Created,
+		"running":    capsule.Running,
+		"paused":     capsule.Invalid, // a container should never be paused
+		"restarting": capsule.Invalid, // a container should never be restarting
```
Comment (RohanDalton) [inline internal/capsule/docker/docker.go:30]: A *capsule* should never be paused/restarting.

File: internal/capsule/docker/engine.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+)
+
+// Engine defines the interface for treating the
+// Docker API as a capsule.Engine.
+type Engine interface {
+	capsule.Engine
+}
+
```
Comment (RohanDalton) [inline internal/capsule/docker/engine.go:20]: What's the point of doing this?

**Review by RohanDalton — APPROVED** [review]

File: internal/capsule/docker/docker.go (hunk 1 above)
Comment (rocketbitz) [reply internal/capsule/docker/docker.go:30]: Well yes, but this was in the docker package, and I was addressing _docker_ states.

File: internal/capsule/docker/engine.go (hunk 2 above)
Comment (rocketbitz) [reply internal/capsule/docker/engine.go:20]: In the event that we want to add docker specific methods, you could cast a `capsule.Engine` to a `docker.Engine` and use them.

**Pull request merged**